tests/org.eclipse.passage.lic.net.tests/src/org/eclipse/passage/lic/net/tests/io/SafePayloadTest.java

```diff
@@ -9,40 +9,54 @@
  *
  * Contributors:
  *     ArSysOp - initial API and implementation
+ *     IILS mbH (Hannes Wellmann) - Harden KeyContent against different line-delimiters
  *******************************************************************************/
 package org.eclipse.passage.lic.net.tests.io;
 
 import static org.junit.Assert.assertEquals;
 import static org.junit.Assert.assertTrue;
 import static org.junit.Assert.fail;
 
-import java.nio.file.Paths;
+import java.util.Arrays;
+import java.util.List;
 
 import org.eclipse.passage.lic.api.LicensingException;
 import org.eclipse.passage.lic.api.io.KeyKeeper;
-import org.eclipse.passage.lic.base.io.FileKeyKeeper;
 import org.eclipse.passage.lic.base.io.MD5Hashes;
 import org.eclipse.passage.lic.internal.net.io.SafePayload;
+import org.junit.ClassRule;
 import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+import org.junit.runners.Parameterized.Parameter;
+import org.junit.runners.Parameterized.Parameters;
 
+@RunWith(Parameterized.class)
 public final class SafePayloadTest {
+	@ClassRule
+	public static TemporaryFolder folder = new TemporaryFolder();
+
+	@Parameter
+	public TestKeyKeeper factory;
+
+	@Parameters
+	public static List<TestKeyKeeper> keyKeepers() {
+		return Arrays.asList(new SimpleKeyKeeper(), new KeyKeeperWithOppositeLineDelimiter());
+	}
 
 	@Test
-	public void symmetric() {
+	public void symmetric() throws Throwable {
 		String original = "S0me sophisticäted Str!ng"; //$NON-NLS-1$
 		try {
-			byte[] encoded = new SafePayload(keerper(), new MD5Hashes()).encode(original.getBytes());
+			KeyKeeper keeper = new SimpleKeyKeeper().get();
+			byte[] encoded = new SafePayload(keeper, new MD5Hashes()).encode(original.getBytes());
 			assertTrue(encoded.length > 0);
-			System.out.println(new String(encoded));
-			byte[] decoded = new SafePayload(keerper(), new MD5Hashes()).decode(encoded);
+			byte[] decoded = new SafePayload(this.factory.get(), new MD5Hashes()).decode(encoded);
 			assertTrue(decoded.length > 0);
 			assertEquals(original, new String(decoded));
 		} catch (LicensingException e) {
```

Comment:
Catch `Throwable` here, as we already have catching logic. 
Print exception trace.

Reply:
Done.
Although it actually is not necessary to catch Exceptions in tests at all. If the test method throws an Exception it fails anyway and modern day JUnit Tooling in the IDE and CI usually provides the StackTrace then any way.